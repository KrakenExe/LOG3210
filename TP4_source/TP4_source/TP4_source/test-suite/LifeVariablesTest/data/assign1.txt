Declaration
num a;
num b;
num c;
num d;
num e;
EndDeclaration

b = a+c;
c = b;
d = c;
e = d;
a = e;