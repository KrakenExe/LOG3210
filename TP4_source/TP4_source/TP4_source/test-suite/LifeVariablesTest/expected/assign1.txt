===== STEP 0 ===== 
REF : [a]
DEF : [b]
IN  : [a]
OUT : [b]
SUCC: [_step1]
PRED: []

===== STEP 1 ===== 
REF : [b]
DEF : [c]
IN  : [b]
OUT : [c]
SUCC: [_step2]
PRED: [_step0]

===== STEP 2 ===== 
REF : [c]
DEF : [d]
IN  : [c]
OUT : [d]
SUCC: [_step3]
PRED: [_step1]

===== STEP 3 ===== 
REF : [d]
DEF : [e]
IN  : [d]
OUT : [e]
SUCC: [_step4]
PRED: [_step2]

===== STEP 4 ===== 
REF : [e]
DEF : [a]
IN  : [e]
OUT : []
SUCC: []
PRED: [_step3]

